rama-cli/src/cmd/tls/mod.rs

```diff
@@ -0,0 +1,158 @@
+        client::{ConnectorService, EstablishedClientConnection},
+        tls::{
+            DataEncoding,
+            client::{ClientConfig, NegotiatedTlsParameters},
+        },
+    },
+    tcp::client::{Request, service::TcpConnector},
+    tls::{
+        rustls::client::{TlsConnectorData, TlsConnectorLayer},
+        std::dep::boring::x509::X509,
+    },
+};
+use tokio::net::TcpStream;
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::{EnvFilter, fmt, layer::SubscriberExt, util::SubscriberInitExt};
+
+#[derive(Args, Debug, Clone)]
+/// rama tls support
+pub struct CliCommandTls {
+    /// The address to connect to
+    /// e.g. "example.com" or "example.com:8443"
+    /// if no port is provided, the default port 443 will be used
+    address: String,
+
+    #[arg(long, short = 'k')]
+    /// Wether to skip certificate verification
+    insecure: bool,
+}
+
+/// Run the tls command
+pub async fn run(cfg: CliCommandTls) -> Result<(), BoxError> {
+    tracing_subscriber::registry()
+        .with(fmt::layer())
+        .with(
+            EnvFilter::builder()
+                .with_default_directive(LevelFilter::INFO.into())
+                .from_env_lossy(),
+        )
+        .init();
+
+    // let authority = Authority::try_from(cfg.address.clone())?;
+
+    let authority = match Authority::try_from(cfg.address.clone()) {
+        Ok(authority) => authority,
+        Err(err) => {
+            // if missing port, we can try to add the default port
+            if err.to_string().contains("missing port") {
+                let authority = format!("{}:443", cfg.address);
+                Authority::try_from(authority)?
+            } else {
+                return Err(err.into());
+            }
+        }
+    };
+
+    let tls_client_data = TlsConnectorData::try_from(ClientConfig {
+        store_server_certificate_chain: true,
+        ..Default::default()
+    })
+    .expect("create tls connector data for client");
+
+    let tcp_connector = TcpConnector::new();
+
+    let loggin_service = LoggingLayer.layer(tcp_connector);
+
+    let tls_connector = TlsConnectorLayer::secure()
+        .with_connector_data(tls_client_data)
+        .layer(loggin_service);
+
+    let ctx = Context::default();
+
+    let EstablishedClientConnection { ctx, req, conn } =
+        tls_connector.connect(ctx, Request::new(authority)).await?;
+
+    let params = ctx.get::<NegotiatedTlsParameters>().unwrap();
+
+    if let Some(cert_chain) = params.peer_certificate_chain.clone() {
+        match cert_chain {
+            DataEncoding::Der(raw_data) => match X509::from_der(&raw_data) {
+                Ok(cert) => {
+                    println!("Certificate: {:?}", cert);
+
+                    println!("Subject: {:?}", cert.subject_name());
+                    println!("Issuer: {:?}", cert.issuer_name());
+                }
+                Err(err) => {
+                    println!("Error decoding certificate: {:?}", err);
```

Comment:
```suggestion
                    eprintln!("Error decoding certificate: {:?}", err);
```

and please also keep track that erorrs have occurred such that you can return an error at the end as well to make sure we exit with a non-success status code due to errors that have happened